tests/syft/lib/allowlist_test.py

```diff
@@ -0,0 +1,495 @@
+        BASIC_OPS_RETURN_TYPE[method_name] = return_type
+
+# load our custom configurations
+with open(__file__.replace(".py", ".json"), "r") as f:
+    TEST_JSON = json.loads(f.read())
+
+
+TEST_DATA = []
+# we iterate over the allowlist and only override where explicitly defined
+for op in BASIC_OPS:
+    skip = []
+    if op not in TEST_JSON["tests"]["torch.Tensor"]:
+        # there is no custom configuration so we will test all supported combinations
+        dtypes = ["common"]
+        test_tensors = ["tensor1", "tensor2"]
+        test_inputs = ["all"]
+        is_property = False
+        return_type = BASIC_OPS_RETURN_TYPE[op]
+    else:
+        meta = {}
+        # if there is a profile use the settings from that first
+        if "profile" in TEST_JSON["tests"]["torch.Tensor"][op]:
+            profile = TEST_JSON["tests"]["torch.Tensor"][op]["profile"]
+            meta.update(TEST_JSON["profiles"][profile])
+
+        # now update with what ever extra settings are overridden on the method
+        meta.update(TEST_JSON["tests"]["torch.Tensor"][op])
+        dtypes = meta["data_types"]
+        test_tensors = meta["tensors"]
+        test_inputs = meta["inputs"]
+        is_property = meta["property"]
+        return_type = meta["return_type"]
+
+        # grab skip rules
+        if "skip" in meta:
+            skip += meta["skip"]
+
+        if "min_version" in meta and TORCH_VERSION < version.parse(meta["min_version"]):
+            # skip attributes which are not supported in the TORCH_VERSION
+            continue
+
+    # gather all the data types we want to test
+    data_types = []
+    for dtype in dtypes:
+        if issubclass(type(dtype), str) and dtype in TEST_JSON["data_types"]:
+            data_types += TEST_JSON["data_types"][dtype]
+        else:
+            data_types.append(dtype)
+
+    # collect the tensors we want to test
+    tensors = []
+    for tensor in test_tensors:
+        if issubclass(type(tensor), str) and tensor in TEST_JSON["tensors"]:
+            tensors += TEST_JSON["tensors"][tensor]
+        else:
+            tensors.append(tensor)
+
+    # get the list of inputs
+    inputs = []
+    for input in test_inputs:
+        if issubclass(type(input), str) and input in TEST_JSON["inputs"]:
+            inputs += TEST_JSON["inputs"][input]
+        else:
+            inputs.append(input)
+
+    combinations = list(
+        product(
+            data_types,
+            [op],
+            tensors,
+            inputs,
+            [is_property],
+            [return_type],
+        )
+    )
+
+    skipped_combinations = set()
+    for combination in combinations:
+        for skip_rule in skip:
+            if check_skip(
+                combination=combination, skip_rule=skip_rule, lib_version=TORCH_VERSION
+            ):
+                # we use str(combination) so that we can hash the entire combination
+                skipped_combinations.add(str(combination))
```

Comment:
Why not directly doing `TEST_DATA.append(combination)` here if the condition is False?

Reply:
Yeah this is a bit annoying. I think my logic was this:
There are multiple skip rules at:
> for skip_rule in skip:
So the possible times we "dont skip" is more than 1, so we would need to use a set to add the combination to but the combination contains a nested list which cant be hashed, so instead we just use another set of just the str representation to keep track of which ones shouldn't be used and then do a second loop for appending the real combination with full intact data structure.

Definitely feels sub optimal, and I would love a suggestion on a good alternative. I considered changing the nested list to a tuple but JSON doesn't have a tuple representation so I figured i'd just go this route for now.

Reply:
So you mean that several items in combinations can be the same combination?

Reply:
It's a nested loop so its combination * skip_rules.
```
for combination in combinations:
    # skips are temporary
    for skip_rule in skip:
```
So appending based on the  check_skip could add more than 1 entry of the same combination.

Reply:
And what if we break the skip loop as soon as the combination needs to be skipped?

Reply:
yup that works:
```
before change
============ 9430 passed, 6355 warnings in 184.55s (0:03:04) ============
after change
============ 9430 passed, 6355 warnings in 190.64s (0:03:10) ============
```

The tests just passed so the next commit will reset the clock for 23 minutes from now, so ill hold off until there are any final requests.